packages/contracts/contracts/libs/v0.8.x/minter-libs/DALib.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Art Blocks Merkle Library
+ * @notice This library is designed to manage and verify merkle based gating for Art Blocks projects.
+ * It provides functionalities such as updating the merkle root of project, verifying an address against a proof,
+ * and setting the maximum number of invocations per address for a project.
+ * @author Art Blocks Inc.
+ */
+
+library DALib {
+    using SafeCast for uint256;
+
+    struct DAProjectConfig {
+        // @dev max uint64 ~= 1.8e19 sec ~= 570 billion years
+        uint64 timestampStart;
+        uint64 timestampEnd;
+        uint64 priceDecayHalfLifeSeconds;
+        uint128 startPrice;
+        uint128 basePrice;
+    }
+
+    function resetAuctionDetailsExp(
+        DAProjectConfig storage _auctionProjectConfigMapping
+    ) external {
+        _auctionProjectConfigMapping.timestampStart = 0;
+        _auctionProjectConfigMapping.priceDecayHalfLifeSeconds = 0;
+        _auctionProjectConfigMapping.startPrice = 0;
+        _auctionProjectConfigMapping.basePrice = 0;
+    }
+
+    function resetAuctionDetailsLin(
+        DAProjectConfig storage _auctionProjectConfigMapping
+    ) external {
+        // reset to initial values
+        _auctionProjectConfigMapping.timestampStart = 0;
+        _auctionProjectConfigMapping.timestampEnd = 0;
+        _auctionProjectConfigMapping.startPrice = 0;
+        _auctionProjectConfigMapping.basePrice = 0;
+    }
+
+    function setAuctionDetailsLin(
```

Comment:
nit: add natspec comments here, with dev comments about minter implementations being required to check for any additional guardrails such as min/max auction length, etc.